cl/corpus_importer/management/commands/update_casenames_wl_dataset.py

```diff
@@ -0,0 +1,368 @@
+
+                # Validate the volume
+                if volume and volume.isdigit():
+                    # Append the validated citation as a dictionary
+                    validated_citations.append(
+                        {
+                            "volume": citation.groups["volume"],
+                            "reporter": citation.corrected_reporter(),
+                            "page": citation.groups["page"],
+                        }
+                    )
+                else:
+                    # If volume is invalid, skip this citation
+                    continue
+
+    return validated_citations
+
+
+def query_possible_matches(
+    valid_citations: list[dict], docket_number: str, date_filed: date
+) -> QuerySet[Citation]:
+    """Find matches for row data
+
+    :param valid_citations: list of FullCaseCitation objects
+    :param docket_number: cleaned docket number from row
+    :param date_filed: formatted filed date from row
+
+    :return: list of matched OpinionCluster
+    """
+
+    citation_queries = Q()
+
+    for citation in valid_citations:
+        citation_query = Q(**citation) & Q(
+            cluster__docket__docket_number__contains=docket_number,
+            cluster__date_filed=date_filed,
+        )
+        citation_queries |= citation_query
+    possible_matches = Citation.objects.filter(
+        citation_queries
+    ).select_related("cluster")
+
+    return possible_matches
+
+
+def update_matched_case_name(
+    matched_cluster: OpinionCluster, csv_case_name: str
+) -> tuple[bool, bool]:
+    """Update case name of matched cluster and related docket if empty any of them
+
+    :param matched_cluster: OpinionCluster object
+    :param csv_case_name: case name from csv row
+    :return: tuple with boolean values if cluster and related docket case name updated
+    """
+    cluster_case_name_updated = False
+    docket_case_name_updated = False
+
+    if not matched_cluster.case_name:
+        # Save case name in cluster when we don't have it
+        matched_cluster.case_name = csv_case_name
+        matched_cluster.save()
+        logger.info(f"Case name updated for cluster id: {matched_cluster.id}")
+        cluster_case_name_updated = True
+
+    if not matched_cluster.docket.case_name:
+        # Save case name in docket when we don't have it
+        matched_cluster.docket.case_name = csv_case_name
+        matched_cluster.docket.save()
+        logger.info(
+            f"Case name updated for docket id: {matched_cluster.docket.id}"
+        )
+        docket_case_name_updated = True
+
+    return cluster_case_name_updated, docket_case_name_updated
+
+
+def combine_initials(case_name: str) -> str:
+    """Combine initials in case captions
+
+    :param case_name: the case caption
+    :return: the cleaned case caption
+    """
+
+    pattern = r"((?:[A-Z]\.?\s?){2,})(\s|$)"
+
+    return re.sub(pattern, lambda m: m.group(0).replace(".", ""), case_name)
```

Comment:
this pattern doesn't appear to removal initials

Reply:
perhaps 

```
initials_pattern = re.compile(r"([A-Z]{1}\.?\s?){2,}(\s|$)")

    match = initials_pattern.search(case_name)  # Search for the initials pattern
    if match:
        initials = match.group()
        compressed_initials = re.sub(r"(?!\s$)[\s\.]", "", initials)
        case_name = case_name.replace(initials, compressed_initials)  
```

this would work?  I tested it on 

case_names = [
    "M. X. Smith",
    "M.X. Smith",
    "M X Smith",
    "M. X. J. Smith",
    "M X J Smith",
]
and it worked as expected .

Reply:
we could have abbreviations in both parties, but that way it will only apply to the first match, we should do something like:

```
def initials_func(case_name):
    initials_pattern = re.compile(r"([A-Z]{1}\.?\s?){2,}(\s|$)")

    matches = initials_pattern.finditer(case_name)  # Search for the initials pattern
    if not matches:
        return case_name
    for match in matches: # replace all abbreviations in the case name
        initials = match.group()
        compressed_initials = re.sub(r"(?!\s$)[\s\.]", "", initials)
        case_name = case_name.replace(initials, compressed_initials)
    return case_name

```
Because if not it fails like this:

```
D.L.M. v. T.J.S. -> DLM v. T.J.S.
R. L. C. R. v. L. Z. S. -> RLCR v. L. Z. S.
J. B. v. C. E. -> JB v. C. E.
```

After applying what I mentioned above, the regex only fails in this specific case:

`In re Gregory C. KAPORDELIS -> In re Gregory CKAPORDELIS
`

Also, I'm not sure if "M X J" in M ​​X J Smith should be considered an abbreviation since they are not together or have periods at the end of each abbreviated word.

I'm checking how to improve the regex, the other changes are ready. I also added a new date format to the list of valid formats.

Reply:
how about something like this 

```
def initials_func(case_name):
	initials_pattern = re.compile(r"(\b[A-Z]{1}\.?\s?){2,}(\s|$)")

	matches = initials_pattern.finditer(case_name)
	if matches:
		for match in matches:
			initials = match.group()
			compressed_initials = re.sub(r"(?!\s$)[\s\.]", "", initials)
			case_name = case_name.replace(initials, compressed_initials)
	return case_name
```